lib/gems/pending/util/miq_winrm.rb

```diff
@@ -1,132 +0,0 @@
-      $log.debug "Error Running Powershell on #{hostname} using user \"#{username}\": #{error}" if $log
-      raise if wmi_error_retries > WMI_RETRIES
-      wmi_error_retries += 1
-      if error.to_s.include? "This user is allowed a maximum number of "
-        $log.debug "Re-opening connection and retrying" if $log
-        @shell.close
-        @shell      = nil
-        @connection = raw_connect(@options)
-        execute
-        retry
-      else
-        raise
-      end
-    end
-  end
-
-  def run_elevated_powershell_script(script)
-    wmi_error_retries = 0
-    begin
-      elevate if @elevated_runner.nil?
-      $log.debug "Running powershell script elevated on #{hostname} as #{username}:\n#{script}" if $log
-      @shell.run(script)
-    rescue WinRM::WinRMAuthorizationError
-      $log.info "Error Logging In to #{hostname} using user \"#{username}\"" if $log
-      raise
-    rescue WinRM::WinRMWMIError => error
-      $log.debug "Error Running Powershell on #{hostname} using user \"#{username}\": #{error}" if $log
-      raise if wmi_error_retries > WMI_RETRIES
-      wmi_error_retries += 1
-      if error.to_s.include? "This user is allowed a maximum number of "
-        $log.debug "Re-opening connection and retrying" if $log
-        @shell.close
-        @connection      = raw_connect(@options)
-        @elevated_runner = nil
-        retry
-      else
-        raise
-      end
-    end
-  end
-
-  # Parse an ugly XML error string into something much more readable.
-  #
-  def parse_xml_error_string(str)
-    require 'nokogiri'
-    str = str.sub("#< CLIXML\r\n", '') # Illegal, nokogiri can't cope
-    doc = Nokogiri::XML::Document.parse(str)
-    doc.remove_namespaces!
-
-    text = doc.xpath("//S").map(&:text).join
-    array = text.split(/_x\h{1,}_/) # Split on stuff like '_x000D_'
-    array.delete('') # Delete empty elements
-
-    array.inject('') do |string, element|
-      break string if element =~ /at line:\d+/i
-      string << element
-    end
-  end
-
-  private
-
-  def validate_options(options)
-    raise "no Username defined" if options[:user].nil?
-    raise "no Password defined" if options[:password].nil?
-    raise "no Hostname defined" if options[:hostname].nil?
-    @username = options[:user]
-    @password = options[:password]
-    @hostname = options[:hostname]
-    @port     = options[:port] unless options[:port].nil?
-  end
-
-  def raw_connect(opts)
-    # HACK: WinRM depends on the gssapi gem for encryption purposes.
-    # The gssapi code outputs the following warning:
-    #   WARNING: Could not load IOV methods. Check your GSSAPI C library for an update
-    #   WARNING: Could not load AEAD methods. Check your GSSAPI C library for an update
-    # After much googling, this warning is considered benign and can be ignored.
-    # Please note - the webmock gem depends on gssapi too and prints out the
-    # above warning when rspec tests are run.
-    # silence_warnings { require 'winrm' }
```

Comment:
Do we need to care about gssapi gem?  I don't see it in the dependency list, so I guess it was a dep of winrm?

Reply:
Note, I saw there were references to kerberos and scvmm providers as if scvmm is the only kerberos type authentication.  I didn't remove that stuff because I wasn't comfortable touching that code when it was more than just scvmm.

I do think there is more to delete but as a followup since this code is gated behind conditions in the UI that cannot happen:

```
app/views/ems_infra/_form_fields.html.haml:  - if @edit[:new][:emstype] == "scvmm" || @edit[:new][:emstype] == "openstack" || @edit[:new][:emstype] == "openstack_infra"
app/views/layouts/_auth_credentials.html.haml:    - if @edit[:new][:emstype] == "scvmm" && @edit[:new][:default_security_protocol] == "kerberos"
app/views/layouts/angular-bootstrap/_auth_credentials_angular_bootstrap.html.haml:        .note{"ng-if" => "emsCommonModel.emstype == 'scvmm' && emsCommonModel.default_security_protocol == 'kerberos'"}
app/views/layouts/angular-bootstrap/_endpoints_angular.html.haml:                            "emsCommonModel.emstype == 'scvmm' || " +                                  |
app/views/layouts/angular-bootstrap/_endpoints_angular.html.haml:    .col-md-8{"ng-if" => "emsCommonModel.emstype == 'scvmm'"}
app/views/layouts/angular-bootstrap/_endpoints_angular.html.haml:                       options_for_select([["<#{_('Choose')}>", nil]] + @scvmm_security_protocols, :disabled => ["<#{_('Choose')}>", nil]),
app/views/layouts/angular-bootstrap/_endpoints_angular.html.haml:            "ng-if" => "emsCommonModel.emstype == 'scvmm' && emsCommonModel.default_security_protocol == 'kerberos'"}
app/views/layouts/angular/_multi_auth_credentials.html.haml:                "#{ng_model}.emstype == 'scvmm'                  || " + |
```